snippets/vectorAngle.md

```diff
@@ -0,0 +1,36 @@
+---
+title: vectorAngle
+tags: math,intermediate
+---
+
+Returns the angle (theta) value between two vectors.
+
+Use `Math.trunc()` to pick the integer value of length. Use `slice()` slice the incoming array into individual vectors. 
+Use `forEach()`, `Math.pow()` and `Math.sqrt()` to iterate over the vectors and calculate the magnitude of each vector and the scalar product.
+Use `Math.acos()` to calculate arccos and get the theta.
+
+
+```js
+const theta = ((...x) =>{ 
+  let half = Math.trunc(x.length / 2);
+  let a = x.slice(0 , half);
+  let b = x.slice(half);  
+
+  let magnitudeOfA = 0;
+  a.forEach((a) => magnitudeOfA += Math.pow(a , 2));
+  magnitudeOfA = Math.sqrt(magnitudeOfA);
```

Comment:
This whole thing could be very easily altered to be a one-liner:
```js
const magnitudeA = Math.sqrt(a.reduce((acc, el) => acc + Math.pow(el, 2), 0));
```

Reply:
Done